### ISSUE TITLE
[Bug] Fixing the lengthscale of a kernel. MaternKernel object has no attribute base_kernel

### ISSUE THREAD

**Issue by aegonwolf** [cornellius-gp/gpytorch#2196]
# 🐛 Question

Hi all, 

I tried this https://github.com/cornellius-gp/gpytorch/issues/689 to fix my lengthscale of a MaternKernel.

_Originally I got an attribute error but that probably was something on my part as it has miraculously vanished._
When I fix the lengthscale as mentioned above and remove the parameter from training the model doesn't learn anything anymore and just outputs the same mean and sigma for the whole domain.

In addition, I would also like to add a fixed variance which I now do with the fixed noise likelihood. Can one use the deprecated White Noise Kernel?

Many thanks in advance!

Sorry for closing and opening again, I saw that there was a question label and I thought it would be more fitting but it seems I can't add it from my end.

**Comment by Balandat** [comment]
Can you provide a code example so we can investigate? 

> In addition, I would also like to add a fixed variance which I now do with the fixed noise likelihood. Can one use the deprecated White Noise Kernel?

Is there a particular reason you want to use the white noise kernel?

**Comment by aegonwolf** [comment]
> Can you provide a code example so we can investigate?
> 
> > In addition, I would also like to add a fixed variance which I now do with the fixed noise likelihood. Can one use the deprecated White Noise Kernel?
> 
> Is there a particular reason you want to use the white noise kernel?

Hi, thanks a lot for answering!

I found the issue after investigating the source code, my kernel is composite, so I have to access via: `base_kernel.kernels[idx of matern]`.

This is for a school problem, I was able to solve it with sklearn and gpy but I'd really like to learn and use gpytorch and botorch.

We are supposed to do safe BO with constraints and the functions are simulated at random but with Matern Kernels that have a fixed variance and lengthscale. In addition a fixed white noise is added.
I did model it with learned parameters and constraints but it does a lot worse than my composite kernels with gpy and sklearn with fixed parameters.

Its no big issue, but I thought for sure there is a way to do it in PyTorch, if there isn't, I'd super appreciate if I'd know though, so I won't spend more time on this.

I tried adding jitter.
